sdk/src/main/java/io/opentelemetry/sdk/trace/config/TraceConfig.java

```diff
@@ -89,6 +93,7 @@
   private static final int DEFAULT_SPAN_MAX_NUM_LINKS = 32;
   private static final int DEFAULT_SPAN_MAX_NUM_ATTRIBUTES_PER_EVENT = 32;
   private static final int DEFAULT_SPAN_MAX_NUM_ATTRIBUTES_PER_LINK = 32;
+  private static final int DEFAULT_KEY_SPAN_ATTRIBUTE_MAX_VALUE_LENGTH = 0;
```

Comment:
Is zero a good default? shouldn't it be MAX_INT?

Reply:
even better, make the default be a null `Integer` and get rid of this magic number.

Reply:
All other methods here return `int`. Do you want for this one to return `Integer`?

With default value being MAX_INT the code will look like `if(traceConfig.getMaxLengthOfAttributeValues() < Integer.MAX_INT) truncate()`, which is strange for me.

Reply:
How about `if(traceConfig.getMaxLengthOfAttributeValues() != TraceConfig.DEFAULT_MAX_ATTRIBUTE_LENGTH)` ? I think that reads great, makes it clear, and doesn't use '0' as a magic number.

Reply:
Also, why is the word 'KEY' in this constant name?

Reply:
Oi, KEY is certainly wrong, will fix.

Do you propose to have `TraceConfig.DEFAULT_MAX_ATTRIBUTE_LENGTH` constant but let it have `null` as a value or `Integer.MAX_VALUE`?

Reply:
I think I prefer MAX_VALUE, to the nullable, but I don't feel super strongly about it.

Reply:
Wait, that does not make sense. Currently I attempt to truncate values if configured value differs from "unlimited" which is signalled by 0. Your proposal says "let's try to truncate if configured value is not default". But what if we change default to, say, 2M? `if(traceConfig.getMaxLengthOfAttributeValues() != TraceConfig.DEFAULT_MAX_ATTRIBUTE_LENGTH)` will not work anymore. Compare with default is wrong, default may change. We have to compare with "unlimited", however we denote that. And I like `traceConfig.getMaxLengthOfAttributeValues() > 0` more than `traceConfig.getMaxLengthOfAttributeValues() > Integer.MAX_INT`.

Reply:
I'd be happier with `-1`, rather than `0`. `0` is actually a valid length, and `-1` clearly is not. And, I'd rather have a constant defined for the "unset" value, rather than just hardcode the magic number in two places.

Reply:
@jkwatson  PTAL

Reply:
👍 I like `-1` for unlimited, I think that's reasonably conventional